pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -412,7 +412,7 @@ func NewController(kubeClient kubelib.Client, options Options) *Controller {
 			})
 		}
 	})
-	c.registerHandlers(c.pods.informer, "Pods", c.pods.onEvent, nil)
+	c.registerHandlers(c.pods.informer, "Pods", c.pods.onEvent, c.pods.checkPodLabelChange)
```

Comment:
why check the labels in filter func

Reply:
Because the handler func does not include the old object, and filter func is the only place to fetch it.

I know it looks kind of hacky to do this in the filter func, but I believe it's way cheaper than maintaining a map just for fetching old pods

Reply:
IC